### PR TITLE
[Feedback wanted] Fix for first launch fail if casper theme is missing.

### LINKED ISSUE #4866: Use first theme available if casper doesn't exist on first run.
Currently on first run, the server crashes if the casper theme is not present.  I suggest using the first theme available if casper doesn't exist and only throwing if no themes are available.

Would this be desirable behavior?  I can submit a PR if so.

### PR DESCRIPTION
**Pull request by olsio**
closes #4866

During validation a check is run if the activeTheme is present and
fails if missing. New logic is to return another possible theme and
updates the current activeTheme to the alternative.

### CODE REVIEW

**Comment by olsio** [comment]
Issues I see:
- Might be mis-using error for the exception flow
- Test missing yet
- Not sure this could hide an error when some one assigns a non-existing theme as activeTheme and it will silently fallback to another theme.

File: core/server/models/settings.js [hunk 1]
```diff
@@ -84,7 +84,9 @@ Settings = ghostBookshelf.Model.extend({
                 return;
             }
 
-            return validation.validateActiveTheme(themeName);
+            return validation.validateActiveTheme(themeName).catch(errors.ActiveThemeNotFoundError, function (error) {
+                return self.save({value: error.fallbackTheme}, {patch: true});
```
Comment (olsio) [inline core/server/models/settings.js]: Not sure if that is the correct way to update the current setting or if I should call Settings.edit or something similar.

**Comment by ErisDS** [comment]
Not ignoring this - thinking about it :wink:

**Comment by olsio** [comment]
Thanks for letting me know. I couldn't find any place that would actually fix pre-launch things so let me know if you have better places to put this logic.

**Comment by kirrg001** [comment]
Thanks for your work @olsio 👋
This PR is pretty old and has many conflicts. Closing it now.
We would be happy about a new PR if you have time - the issue is still open.

**Pull request closed without merging**